nix/default.nix

```diff
@@ -24,10 +24,11 @@ import sources.nixpkgs {
       import ./scripts.nix {
         inherit pkgs;
         config = {
-          chainmain-config = ../scripts/chainmain-devnet.yaml;
-          cronos-config = ../scripts/cronos-devnet.yaml;
-          hermes-config = ../scripts/hermes.toml;
-          geth-genesis = ../scripts/geth-genesis.json;
+          chainmain-config = builtins.toString ../scripts/chainmain-devnet.yaml;
```

Comment:
Why the `builtins.toString` call?

Reply:
It is necessary to get the full path of the file (string), instead of copying to nix store. https://github.com/crypto-org-chain/cronos/blob/main/nix/scripts.nix#L15:  if we use  `./scripts/cronos-devnet.yaml`, then quote it with  `${config.cronos-config}` will copy to nix store. 

Not copying to nix sotre has following benefits:
1. Read `.env` with relative path just under the project `script` folder.
2. In nix-shell environment, if we modify the `cronos-devnet.yaml`, we can call start-cronos directly without exiting the nix-shell and re-enter the nix-shell.

Reply:
Before:
```
#!/nix/store/c9cxq1583a85bsq76q2rbnbiwwp7ygxr-bash-4.4-p23/bin/bash
# rely on environment to provide cronosd
export PATH=/nix/store/f347n055kzsyq30m41x481dpsq0zbip1-python3.8-pystarport-0.2.2/bin:$PATH
source /nix/store/ym2jzjqjmzf7azjkc8m9jd4vy0qjrx5p-dotenv
/nix/store/zhicf2wijlr36315xx1nxq5bhh6q90qs-start-cronos /nix/store/k0mfjv85j0sjbfjj6lisriwq3pkzcsqi-cronos-devnet.yaml $@

```


After:

```
#!/nix/store/c9cxq1583a85bsq76q2rbnbiwwp7ygxr-bash-4.4-p23/bin/bash
# rely on environment to provide cronosd
export PATH=/nix/store/2kfp7bbr155h88zy0kglji3ff33f0p4k-python3.8-pystarport-0.2.2/bin:$PATH
source /nix/store/ym2jzjqjmzf7azjkc8m9jd4vy0qjrx5p-dotenv
/nix/store/zhicf2wijlr36315xx1nxq5bhh6q90qs-start-cronos /Users/damonchen/Documents/cronos/scripts/cronos-devnet.yaml $@

```

Reply:
Then you don't get the reproducibility with these scripts.
In development, you can always call `./scripts/start-cronos config data` explicitly.

Reply:
I see, treat it like a fixed binary. Let me think a more thoughtful idea.

Reply:
One question here: since `config` is fixed, only data directory can be changed. Should be `./scripts/start-cronos data` can be called, rather than `./scripts/start-cronos config data`?

Reply:
```
[nix-shell:~/Documents/cronos/integration_tests]$ start-cronos
No data directory supplied
```

Reply:
Oh, seems you make it wrong? `./scripts/start-cronos config data` is different than `start-cronos` binary. If user calls `./scripts/start-cronos config data`, it is always reproducible.

Reply:
It is ok to make `start-cronos` also reproducible as well.

Reply:
But seems no documents here, user would not know `start-cronos` exists. The binary is not useful because it only works in `nix-shell`. Even me, I don't use it at all.

Reply:
Of course, it is important in integration test. Just not that useful for users, assuming they do not have `nix` installed.

Reply:
`start-cronos` and `start-geth` are only used to start cronos and geth devnets with the identical environment, to run cronos devnet with custom config, one can always use `pystarport` directly, the `scripts/start-cronos` is only a thin wrapper over that.

Reply:
Two TODOs here:
1. Make all `ShellScriptBin` reproducible. And add documents if possible.
2. Improve all scripts. E.g. So that user knows they can use `dotenv` option as well.

Reply:
We can remove `start-chainmain` and `start-cronos`, and simply use `pystarport` directly.